src/lib/ProjectM36/Client/Simple.hs

```diff
@@ -0,0 +1,79 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+module ProjectM36.Client.Simple (
+  withDBConnection,
+  withTransaction,
+  executeDatabaseContextExpr,
+  executeRelationalExpr,
+  Atom(..),
+  AtomType(..),
+  DBError(..),
+  Attribute(..),
+  C.ConnectionInfo(..),
+  C.PersistenceStrategy(..),
+  C.NotificationCallback(..),
+  C.emptyNotificationCallback,
+  C.DatabaseContextExpr(..),
+  C.RelationalExprBase(..)  
+  ) where
+-- | A simplified, monad-based client for Project:M36 database access.
+import ProjectM36.Error
+import ProjectM36.Base
+import qualified ProjectM36.Client as C
+import Control.Monad.Reader
+import Control.Exception.Base
+
+-- | A union of connection and other errors that can be returned from 'withDBConnection'.
+data DBError = ConnError C.ConnectionError |
+               RelError RelationalError
+               deriving (Eq, Show)
+
+newtype DB a = DB {runDB :: ReaderT (C.SessionId, C.Connection) IO a}
+  deriving (Functor, Applicative, Monad, MonadIO)
+
+-- | Opens a connection in the 'DB' monad and closes it after running the monad.
+withDBConnection :: C.ConnectionInfo -> DB (Either DBError a) -> IO (Either DBError a)
+withDBConnection connInfo block = bracket (C.connectProjectM36 connInfo) (\eConn -> case eConn of
+                                                                             Left err -> pure (Right ())
+                                                                             Right conn -> C.close conn >> pure (Right ())) $ \eConn -> do
+  case eConn of
+    Left err -> pure (Left (ConnError err))
+    Right conn -> do
+      eSess <- C.createSessionAtHead conn "master"
+      case eSess of
+        Left err -> pure (Left (RelError err))
+        Right session ->
+           runReaderT (runDB block) (session, conn)
+           
+wrapClientIO :: (C.SessionId -> C.Connection -> IO (Either RelationalError a)) -> DB (Either DBError a)
+wrapClientIO func = DB $ do
+  (sess, conn) <- ask
+  ret <- liftIO (func sess conn)
+  case ret of
+    Left err -> pure (Left (RelError err))
+    Right val -> pure (Right val)
+           
+-- | Atomically commits the result of the database context expression to disk on the server or restores the initial state.
+withTransaction :: C.DatabaseContextExpr -> DB (Either DBError ())           
```

Comment:
I don't quite get what this does. It only accepts a single expr and then does auto-merge-to-head. I would expect the merge-to-head part to happen in `withDBConnection` at the end.